Services/Hubs/BaseHub.cs

```diff
@@ -50,19 +52,31 @@ public async override Task OnConnectedAsync()
 		IDistributedCache cache = serviceProvider.GetRequiredService<IDistributedCache>();
 		AppDbContext dbContext = serviceProvider.GetRequiredService<AppDbContext>();
 
-		string cacheKey = string.Format(CacheKeys.ConnectionEstablished, Context.User.Id());
+		ClaimsPrincipal? user = Context.User;
+
+		if (user == null)
+			return;
+
+		int receiverId = user.Id();
+		string cacheKey = string.Format(CacheKeys.ConnectionEstablished, receiverId);
 		await cache.SetAsync(cacheKey, true, DateTime.UtcNow.AddDays(1));
 
-		int receiverId = Context.User.Id();
 		DateTime now = DateTime.UtcNow;
 
 		MessageEntity[] undeliveredMessages =
-			await dbContext.Messages.Where(m => m.Chat.Users.Select(u => u.Id).Contains(receiverId) && !m.MessageStatusEntities.Any(ms => ms.ReceiverId == receiverId && ms.Status == 1)).ToArrayAsync();
-		MessageStatusEntity[] messageStatusEntities = undeliveredMessages.Select(m => new MessageStatusEntity(m.Id, receiverId, 1, now)).ToArray();
+			await dbContext.Messages
+			.Where(m =>
+				m.Chat.Users.Select(u => u.Id).Contains(receiverId)
+				&& m.SenderId != receiverId
+				&& !m.MessageStatusEntities.Any(ms => ms.ReceiverId == receiverId
+				&& ms.Status == (int)MessageStatus.Received && ms.StatusUpdateDeliveryConfirmed))
```

Comment:
Why is status int? Let's use enums. Or maybe I'm missing something because I'm on the phone?

PS formatting - the any()'s new line continuation should at least have some indentation.